CMakeLists.txt

```diff
@@ -208,6 +208,7 @@ if(CMAKE_COMPILER_IS_GNU)
       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-overflow=2 -Wformat-truncation")
     endif()
     set(CMAKE_C_FLAGS_RELEASE     "-O2")
+    set(CMAKE_C_FLAGS_TSAN        "-fsanitize=thread -g3 -O2")
```

Comment:
Any particular reason to continue on a thread sanitizer error, whereas we abort on an address sanitizer error?

It kind of makes sense to me in that an ASan error may well be heap corruption and it's meaningless to keep going. Tsan errors are more likely to have corruption that's localized in a single object. But we also stop with ASan on likely-recoverable errors such as null pointer dereferences and integer overflows. (Maybe we do that because it was harder than choosing, especially because sanitizer options (used to) differ between gcc and clang.)

Reply:
I have found the continue on error useful - when it documents race conditions for example, there is not necessarily any incorrect behaviour, so no real need to actually stop. It does potentially give more information as well. Having read the docs very carefully, I am not 100% sure there is a 'stop on error' option like ASan / MemSan.